batch/batch/worker/worker.py

```diff
@@ -664,18 +734,39 @@ async def run_container(self) -> bool:
             async with async_timeout.timeout(self.timeout):
                 with open(self.log_path, 'w') as container_log:
                     log.info(f'Creating the crun run process for {self}')
+                    pid_file = f'{self.container_scratch}/container.pid'
+
                     self.process = await asyncio.create_subprocess_exec(
                         'crun',
                         'run',
                         '--bundle',
                         f'{self.container_overlay_path}/merged',
                         '--config',
                         f'{self.config_path}/config.json',
+                        '--pid-file',
+                        pid_file,
                         self.container_name,
                         stdout=container_log,
                         stderr=container_log,
                     )
-                    await self.process.wait()
+
+                    async def wait_for_pid_file():
+                        delay = 0.01
+                        while not os.path.exists(pid_file):
+                            delay = await sleep_and_backoff(delay)
```

Comment:
Does the pid file not immediately exist after `crun run` returns? I believe `crun state` will also print the pid.

Reply:
crun's tests appear to immediately read the pid file after `crun run` returns.

https://github.com/containers/crun/blob/2d58f66d931cca0454fbd10529665922112bb98f/tests/test_pid_file.py#L27-L31

Reply:
I just tried this and am getting file not found error. Will revert back to my original code.

Reply:
Ah, sure, we don't wait for the crun run process to exit so of course there's a race. OK.